### ISSUE TITLE
Rewrite PhenoDigm evidence parser to pySpark

### ISSUE THREAD

**Issue by DSuveges** [opentargets/platform#1403]
For performance issues at some point it needs to be done.  Consider the following:

- Contact IPMC team to find a more efficient way to access data than the solr index.
- When implementing the parser, consider #859

**Comment by andrewhercules** [comment]
As noted in #1473, we will use the `biologicalModelId` value to generate a link to the MGI website.

It was also mentioned that we could display the mouse gene name in the PhenoDigm evidence data table. Upon reviewing the GraphQL API response, the front-end only has access to the mouse gene ID through the `targetInModel` value - see sample query for [IL10RA ad inflammatory bowel disease](https://api-beta-dot-open-targets-eu-dev.appspot.com/api/v4/graphql/browser?query=query%20phenodigmData%20{%20disease(efoId:%20%22EFO_0003767%22)%20{%20evidences(ensemblIds:[%22ENSG00000110324%22],%20datasourceIds:[%22phenodigm%22]){%20count%20rows{%20targetInModel%20targetFromSource%20targetFromSourceId%20targetModulation%20}%20}%20}%20}) (e.g. [ESR1](https://beta.targetvalidation.org/target/ENSG00000091831)).

Would it be possible to also expose the gene name/label as we do with the Mouse Phenotypes data on the target profile page? If so, we could use the same approach used for `targetFromSource` (name/label) and `targetFromSourceId` (id) and the front-end could construct links to Ensembl.

**Comment by d0choa** [comment]
We could change slightly the schema to have something like:
- `targetInModelId` = `ENSMUSG00000026580`
- `targetInModel` = `Selp`

On a similar note, the current `biologicalModelId` field has some unpleasant surprises. The schema needs to be tightened and the absence of information encoded as `null` instead of strange strings.

@andrewhercules this will affect FE in the current evidence

```python
>>> phenodigm.filter(~F.col("biologicalModelId").startswith("MGI")).count()
4273

>>> phenodigm.filter(~F.col("biologicalModelId").startswith("MGI")).select("targetId", "diseaseId", "biologicalModelId").show(10, False)
+---------------+---------------+-----------------------+
|targetId       |diseaseId      |biologicalModelId      |
+---------------+---------------+-----------------------+
|ENSG00000150676|Orphanet_101997|NOT-RELEASED-1365f850b4|
|ENSG00000125944|MONDO_0030015  |NOT-RELEASED-87b389a499|
|ENSG00000138684|Orphanet_289902|NOT-RELEASED-e84003300a|
|ENSG00000185515|Orphanet_255132|NOT-RELEASED-17108379f3|
|ENSG00000130741|Orphanet_3337  |NOT-RELEASED-bf4df353ad|
|ENSG00000143819|Orphanet_355   |NOT-RELEASED-0c4103fa4b|
|ENSG00000100142|Orphanet_656   |NOT-RELEASED-f7a583a70c|
|ENSG00000204334|Orphanet_486   |NOT-RELEASED-ff96ec2653|
|ENSG00000114790|Orphanet_79124 |NOT-RELEASED-37b85d4dc2|
|ENSG00000187288|Orphanet_90635 |NOT-RELEASED-478d4f4b05|
+---------------+---------------+-----------------------+
only showing top 10 rows
```

**Comment by andrewhercules** [comment]
Okay I will update #1473 with a note about the `null` values
